### DISCUSSION TITLE
No connection between the wallet and my react native app.

### DISCUSSION THREAD

**Discussion by ShuraRB** [WalletConnect/walletconnect-monorepo#2852] [Developer Support]
When I open the modal and select a wallet it works fine, it takes me to the wallet app and asks for permission to connect my wallet. When I press connect, it redirects me to my react native application, but no connection returns. It seems that the app is restarted instead of going into the background to receive the connection authorization. How can I fix that part? Or are you working on a solution?

**Comment by ignaciosantise** [comment]
Hey @ShuraRB 👋  More than happy to help however can you please provide us with more detailed reporting?

1) Name and Version of the package you are using
2) Reproducible code via a codeBox
3) Which wallet (platform and version) you used to get to this error
4) Did you check [our example dapp](https://github.com/WalletConnect/react-native-examples/tree/main/dapps/v2Explorer) to compare your implementation?
5) How many users do you know are impacted by this

**Reply by ShuraRB** [reply]
1. Name and Version of the package you are using
    "@walletconnect/modal-react-native": "^1.0.0-rc.1",
3. Which wallet (platform and version) you used to get to this error
I tried with the last version of rainbow wallet, trust wallet, In android I tried and it works fine but in IOS is where it gives me the error.
4. Did you check [our example dapp](https://github.com/WalletConnect/react-native-examples/tree/main/dapps/v2Explorer) to compare your implementation?
Yes
5. How many users do you know are impacted by this
All users that use IOS of my  App

**Comment by ignaciosantise** [comment]
@ShuraRB thanks for the info. Both Rainbow and Trust wallets work with our example dapp. Can you provide a repository so we can reproduce the issue?
